tests/test_prescan_sub.py

```diff
@@ -0,0 +1,348 @@
+            # Subtract bias and bias offset and get cosmic mask
+            iit_images.append(frameobj.image_bias0) # Science area
+            iit_frames.append(frameobj.frame_bias0) # Full frame
+            
+        if len(dataset) != 2:
+            raise Exception(f"Mock dataset is an unexpected length ({len(dataset)}).")
+        
+        for return_full_frame in [True, False]:
+            output_dataset = prescan_biassub(dataset, return_full_frame=return_full_frame)
+
+            output_shape = output_dataset[0].data.shape
+            if output_shape != shapes[obstype][return_full_frame]:
+                raise Exception(f"Shape of output frame for {obstype}, return_full_frame={return_full_frame} is {output_shape}, \nwhen {shapes[obstype][return_full_frame]} was expected.")
+    
+            # check that data, err, and dq arrays are consistently modified
+            dataset.all_data[0, 0, 0] = 0.
+            if dataset[0].data[0, 0] != 0. :
+                raise Exception("Modifying dataset.all_data did not modify individual frame data.")
+
+            dataset[0].data[0,0] = 1.
+            if dataset.all_data[0,0,0] != 1. :
+                raise Exception("Modifying individual frame data did not modify dataset.all_data.")
+
+            dataset.all_err[0, 0, 0] = 0.
+            if dataset[0].err[0, 0] != 0. :
+                raise Exception("Modifying dataset.all_err did not modify individual frame err.")
+
+            dataset[0].err[0,0] = 1.
+            if dataset.all_err[0,0,0] != 1. :
+                raise Exception("Modifying individual frame err did not modify dataset.all_err.")
+
+            dataset.all_dq[0, 0, 0] = 0.
+            if dataset[0].dq[0, 0] != 0. :
+                raise Exception("Modifying dataset.all_dq did not modify individual frame dq.")
+
+            dataset[0].dq[0,0] = 1.
+            if dataset.all_dq[0,0,0] != 1. :
+                raise Exception("Modifying individual frame dq did not modify dataset.all_dq.")
+            
+            corgidrp_result = output_dataset[0].data
+            iit_result = iit_frames[0] if return_full_frame else iit_images[0]
+
+            if not corgidrp_result == approx(iit_result):
+                raise Exception(f"corgidrp result does not match II&T result for generated mock data, obstype={obstype}, return_full_frame={return_full_frame}.")
+
+            # Plot for debugging
+            # fig,axes = plt.subplots(1,3,figsize=(10,4))
+
+            # im0 = axes[0].imshow(corgidrp_result,cmap='seismic')
+            # plt.colorbar(im0)
+            # axes[0].set_title('corgidrp result')
+
+            # im1 = axes[1].imshow(iit_result,cmap='seismic')
+            # plt.colorbar(im1)
+            # axes[1].set_title('II&T result')
+
+            # im2 = axes[2].imshow(corgidrp_result-iit_result,cmap='seismic')
+            # plt.colorbar(im2)
+            # axes[2].set_title('Difference')
+
+            # plt.tight_layout()
+            # plt.show()
+
+
```

Comment:
Note: tolerance for this test in II&T code was set to 1, but I'm not sure how it ever passed given the sigma on the normal distribution was also set to 1. tolerance of 6 seems to pass more regularly

Reply:
Update: they were returning the bias itself as measured by the bias subtraction function, whereas I was only returning the bias-subtracted data. Since each pipeline step should only return a dataset object, I added a header keyword `'MED_BIAS'` which is the median bias measured (including the bias offset) in the step, so this can be referenced in the future. The test works now with the original (~1%) tolerance!